forage-android/src/main/java/com/joinforage/forage/android/network/model/ForageApiResponse.kt

```diff
@@ -5,16 +5,29 @@ import org.json.JSONObject
 sealed class ForageApiResponse<out T> {
     data class Success<out T>(val data: T) : ForageApiResponse<T>()
 
-    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>()
+    data class Failure(val errors: List<ForageError>) : ForageApiResponse<Nothing>() {
+        companion object {
+            fun fromSQSError(sqsError: SQSError): Failure {
+                val forageError = ForageError(
+                    sqsError.statusCode,
+                    sqsError.forageCode,
+                    sqsError.message,
+                    sqsError.details
+                )
+                return Failure(listOf(forageError))
+            }
+        }
+    }
 }
 
 data class ForageError(
     val httpStatusCode: Int,
     val code: String,
-    val message: String
+    val message: String,
+    val details: ErrorMessageDetails? = null
 ) {
     override fun toString(): String {
-        return "Code: $code\nMessage: $message\nStatus Code: $httpStatusCode"
+        return "Code: $code\nMessage: $message\nStatus Code: $httpStatusCode\nError Details (below):\n$details"
```

Comment:
@dleis612 changed the `.toString()` here just fyi

Reply:
If this is causing issues, can we revert this line to the original `toString()` and display the insufficient funds details separately in the sample app?

Reply:
Nice call. Making a separate <TextView /> to display the `snapResponseError` and the `cashResponseError` sounds like the way to go 👍. With any luck the qa-tests will be resolved now:
1. First issue that caused the QA-tests to fail was that I displayed `forageError.toString()` instead of `forageError.message` (mobile-qa-tests expect that later)
2. Second issue that caused the QA-tests to fail was wrapping the PaymentCapture view in a Scroll view since the Cash payment get's cut off. I'm putting [this scroll-view work](https://linear.app/joinforage/issue/FX-548/android-update-payment-capture-view-to-have-scroll-view) to clean up since it will require a PR to the mobile-qa-tests which doesn't feel like a priority right now